go-controller/pkg/cni/cni.go

```diff
@@ -74,17 +74,21 @@ func (pr *PodRequest) cmdAdd() ([]byte, error) {
 	if namespace == "" || podName == "" {
 		return nil, fmt.Errorf("required CNI variable missing")
 	}
-
-	clientset, err := util.NewClientset(&config.Kubernetes)
-	if err != nil {
-		return nil, fmt.Errorf("could not create kubernetes clientset: %v", err)
+	clientSet := pr.KClient
+	if clientSet == nil {
+		var err error
```

Comment:
L78. Not sure if `clientSet` can be `nil`? The KClient passed down from Start() function is guaranteed to be non-nil, right?

Reply:
In the previous implementation, cnishim.go#L121 is where we passed in a nil. The check was to handle that case. As @dcbw suggested, however, it's better to just pass in the clientset via the `HandleCNIRequest`. Made changes to pass the `clientset` in that way.